app/src/main/java/ai/elimu/appstore/synchronization/AppListAdapter.java

```diff
@@ -344,20 +347,25 @@ public void run() {
                                                     public void onWriteToFileDone(final Integer fileSizeInKbsDownloaded,
                                                                                   final String tempApkDir,
                                                                                   final String apkName) {
-                                                        // Hide progress indicators
+                                                        Timber.i("onWriteToFileDone");
+
                                                         uiHandler.post(new Runnable() {
                                                             @Override
                                                             public void run() {
+                                                                // Hide progress indicators
                                                                 holder.progressBarDownload.setProgress(0);
                                                                 holder.textDownloadProgress.setText("");
                                                                 holder.progressBarDownload.setVisibility(View.GONE);
                                                                 holder.textDownloadProgress.setVisibility(View.GONE);
 
-                                                                if ((fileSizeInKbsDownloaded == null) ||
-                                                                        (fileSizeInKbsDownloaded <= 0)) {
-                                                                    downloadCompleteCallback.onDownloadFailed(fileSizeInKbsDownloaded);
-                                                                } else {
+                                                                File downloadedApkFile = new File(tempApkDir, apkName);
```

Comment:
We already verified APK's checkSum in writeResponseBodyToDisk.
If checkSum is invalid, fileSize = -1 will be returned.